experimental/Posets/src/Poset.jl

```diff
@@ -0,0 +1,199 @@
+# Poset
+
+struct Poset
+  cov::Matrix{Int} # covering relations of the poset
+  rel::BitMatrix # general relations
+  set::BitMatrix # indicates whether a general relation was computed
+  elems::Vector{Symbol} # symbols to use for the elements
+end
+
+function Base.length(P::Poset)
+  return length(P.elems)
+end
+
+function Base.show(io::IO, P::Poset)
+  print(io, "poset with $(length(P)) elements")
+end
+
+# PosetElem
+
+struct PosetElem
+  i::Int
+  parent::Poset
```

Comment:
```suggestion
  parent::Poset
  i::Int
```
I think in most structs we have the parent as the first field. But no strong preference here.